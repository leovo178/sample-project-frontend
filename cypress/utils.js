/* eslint-disable no-undef */
const generateRandomNumber = (min, max) => {
	min = Math.ceil(min)
	max = Math.floor(max)
	return Math.floor(Math.random() * (max - min)) + min
}

const clickItemByTestId = testId =>
	cy
		.findByTestId(testId)
		.should('exist')
		.click()

const mockMessage = 'mockmessage'

export { generateRandomNumber, clickItemByTestId, mockMessage }
