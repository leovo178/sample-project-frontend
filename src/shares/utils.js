const AUTH_TOKEN = 'AUTH_TOKEN'

const getToken = () => localStorage.getItem(AUTH_TOKEN)
const setToken = token => localStorage.setItem(AUTH_TOKEN, token)
const deleteToken = () => localStorage.removeItem(AUTH_TOKEN)

const validateEmail = email => {
	const emailRegex = /^[a-z][a-z0-9_\.]{4,32}@[a-z0-9]{2,}(\.[a-z0-9]{2,4}){1,2}$/gm
	return email.match(emailRegex)
}

const validateName = name => {
	const nameRegex = /^[a-zA-Z0-9]{3,30}$/
	return name.match(nameRegex)
}

const validatePassword = password => {
	const passwordRegex = /^[a-zA-Z0-9]{3,}$/
	return password.match(passwordRegex)
}

export {
	getToken,
	setToken,
	deleteToken,
	validateEmail,
	validateName,
	validatePassword,
}
