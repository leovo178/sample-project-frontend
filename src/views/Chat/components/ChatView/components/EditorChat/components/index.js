export { default as SendReservationDialog } from './SendReservationDialog'
