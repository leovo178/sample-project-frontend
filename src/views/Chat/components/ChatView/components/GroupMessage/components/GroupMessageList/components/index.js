export { default as ImageMessage } from './ImageMessage'
export { default as LastCardIndicator } from './LastCardIndicator'
export { default as FileMessage } from './FileMessage'
export { default as TextMessage } from './TextMessage'
export { default as ReservationMessage } from './ReservationMessage'
