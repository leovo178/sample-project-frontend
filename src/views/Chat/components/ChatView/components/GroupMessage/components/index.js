export { default as GroupBreak } from './GroupBreak'
export { default as GroupMessageList } from './GroupMessageList'
