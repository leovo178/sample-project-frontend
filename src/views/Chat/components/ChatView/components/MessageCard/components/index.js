export { default as FileMessage } from './FileMessage'
export { default as ImageMessage } from './ImageMessage'
export { default as LastCardIndicator } from './LastCardIndicator'
