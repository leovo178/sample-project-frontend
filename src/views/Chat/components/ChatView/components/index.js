export { default as EditorChat } from './EditorChat'
export { default as GroupMessage } from './GroupMessage'
export { default as Header } from './Header'
export { default as MessageCard } from './MessageCard'
export { default as PickerEmoji } from './PickerEmoji'
export { default as ViewMessage } from './ViewMessage'
export { default as MessageReservation } from './MessageReservation'
