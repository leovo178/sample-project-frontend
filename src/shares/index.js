import { getToken, setToken, deleteToken } from './utils'

export { getToken, setToken, deleteToken }
