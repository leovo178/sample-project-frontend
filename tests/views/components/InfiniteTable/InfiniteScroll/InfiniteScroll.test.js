import React from 'react'

import { cleanup, render } from '@testing-library/react'

import { InfiniteScroll } from '@views_components/InfiniteTable/components'

import { mockUserList, mockUserTableHeader } from '@tests/shares/utils'

describe('InfiniteScroll', () => {
	const mockProps = {
		hasNextPage: true,
		isNextPageLoading: true,
		isIconClose: false,
		items: mockUserList,
		columns: mockUserTableHeader,
		selectedRow: {},
		loadNextPage: jest.fn(),
		onClickRow: jest.fn(),
		handleDeleteRow: jest.fn(),
	}

	afterEach(() => {
		cleanup()
	})

	it('should match snapshot', () => {
		const { container } = render(<InfiniteScroll {...mockProps} />)

		expect(container).toMatchSnapshot()
	})
})
