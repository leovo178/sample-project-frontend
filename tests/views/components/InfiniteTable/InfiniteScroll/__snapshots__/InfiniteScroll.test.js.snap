// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfiniteScroll should match snapshot 1`] = `
<div>
  <div
    style="overflow: visible; height: 0px; width: 0px;"
  />
  <div
    class="resize-triggers"
  >
    <div
      class="expand-trigger"
    >
      <div
        style="width: 1px; height: 1px;"
      />
    </div>
    <div
      class="contract-trigger"
    />
  </div>
</div>
`;
